File: examples/python/basic_example.py
"""
Chat402 - Basic Python Example

This example shows how to make a simple request to the Chat402 API
"""

import os
import requests

API_URL = 'https://api.chat402.xyz/api/v1'
API_KEY = os.getenv('CHAT402_API_KEY')  # Set this in your environment

def chat(prompt: str, model: str = 'gpt-3.5-turbo') -> dict:
    """Send a prompt to Chat402 API"""
    try:
        response = requests.post(
            f'{API_URL}/prompt',
            json={
                'model': model,
                'prompt': prompt
            },
            headers={
                'Authorization': f'Bearer {API_KEY}',
                'Content-Type': 'application/json'
            }
        )

        if response.status_code == 402:
            raise Exception('Insufficient balance - please top up your wallet')

        response.raise_for_status()
        return response.json()

    except requests.exceptions.RequestException as e:
        raise Exception(f'API request failed: {str(e)}')


def main():
    """Example usage"""
    try:
        result = chat('What is Ethereum?')

        print(f"Response: {result['text']}")
        print(f"Cost: ${result['cost']['totalCost']:.6f}")
        print(f"Tokens: {result['usage']['totalTokens']}")

    except Exception as e:
        print(f'Error: {str(e)}')


if __name__ == '__main__':
    main()

File: examples/nodejs/basic-example.js
/**
 * Chat402 - Basic Node.js Example
 *
 * This example shows how to make a simple request to the Chat402 API
 */

const axios = require('axios');

const API_URL = 'https://api.chat402.xyz/api/v1';
const API_KEY = process.env.CHAT402_API_KEY; // Set this in your environment

async function chat(prompt, model = 'gpt-3.5-turbo') {
  try {
    const response = await axios.post(`${API_URL}/prompt`, {
      model,
      prompt
    }, {
      headers: {
        'Authorization': `Bearer ${API_KEY}`,
        'Content-Type': 'application/json'
      }
    });

    return response.data;
  } catch (error) {
    if (error.response?.status === 402) {
      throw new Error('Insufficient balance - please top up your wallet');
    }
    throw error;
  }
}

// Example usage
async function main() {
  try {
    const result = await chat('What is Bitcoin?');

    console.log('Response:', result.text);
    console.log('Cost:', `$${result.cost.totalCost.toFixed(6)}`);
    console.log('Tokens:', result.usage.totalTokens);
  } catch (error) {
    console.error('Error:', error.message);
  }
}

main();
